lib/src/rules/avoid_types_on_closure_parameters.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.avoid_annotating_types_on_function_expressions;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Avoid annotating types on function expression parameters.';
```

Comment:
The specification refers to either a "function expression" (9 times) or "function literal" (16 times) as an unnamed function written as a literal in code. (In other words, both terms are used interchangeably). A function expression is one kind of "closure", a "tear-off" being another kind of closure. Outside the specification, all three terms appear to be in use, pretty much interchangeably. I'd probably leave it as "function expression" because that's what's in the style guide, and it might be better to change the name to be consistent (though I like "closure" because it's shorter).